src/main/host/syscall/handler/eventfd.rs

```diff
@@ -75,13 +76,20 @@ impl SyscallHandler {
         let mut desc = Descriptor::new(CompatFile::New(OpenFile::new(File::EventFd(file))));
         desc.set_flags(descriptor_flags);
 
-        let fd = ctx
+        let fd = match ctx
             .process
             .descriptor_table_borrow_mut()
-            .register_descriptor(desc);
+            .register_descriptor(desc)
+        {
+            Ok(x) => x,
+            Err(desc) => {
+                CallbackQueue::queue_and_run(|cb_queue| desc.close(ctx.host, cb_queue));
```

Comment:
I forget, is there a reason we don't have Descriptors close themselves on Drop? Or maybe warn and/or debug_panic if they're dropped without closing?

Reply:
Good point. It used to be the case that you had to close them manually, and it would warn you if you didn't. But I forgot that I changed this. So a `Descriptor` holds an `OpenFile`, and the `OpenFile` will automatically close the ref-counted `File` if there are no more `OpenFile`s for that `File`. So it's actually fine to just drop the `Descriptor`, we'll just lose any errors returned by the file's close method, but we're ignoring those errors anyway.

https://github.com/shadow/shadow/blob/34fe9d83186fc084257cd3a50e681db754b28fd3/src/main/host/descriptor/mod.rs#L518-L521

I'll fix up the PR to remove the stuff about closing the descriptor.

Reply:
Actually, this isn't entirely true. For `LegacyFile`s the `Descriptor` holds a `LegacyFileCounter`, which does not close the file when dropped ("This is similar to an `OpenFile` object, but does not close the file when dropped since we don't have a pointer to the host.").

https://github.com/shadow/shadow/blob/34fe9d83186fc084257cd3a50e681db754b28fd3/src/main/host/descriptor/mod.rs#L737-L743

We should be able to just get the `Host` from the worker and close it during the drop. I'll do this in a follow-up PR though.

Reply:
Sounds good. I guess that means we're potentially "leaking" the `LegacyFileCounter` if registering a `LegacyFile` fails, but we were already doing that before this PR as well

Reply:
Yeah, we're not leaking them in the refcount sense, but they won't be cleaned up properly before the refcount reaches zero. I'll post a PR shortly for improving this.